CHANGELOG/v1.7.0.md

```diff
@@ -0,0 +1,433 @@
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.26.x -> v1.29.x
+- Workload Cluster: v1.24.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+
+## Highlights
+
+* Set MachinePool feature flag default to true + Beta
+* Enable kubeadm feature gates mutation
+* Add public function to create new CRD migrator
+* Add MachineSetReady condition to MachineDeployment
+
+## Changes since v1.6.0
+## :chart_with_upwards_trend: Overview
+- 339 new commits merged
+- 5 breaking changes :warning:
+- 12 feature additions ✨
+- 48 bugs fixed 🐛
+
+## :memo: Proposals
+- MachinePool: Update MachinePool Machines proposal with diagrams (#9664)
+
+## :warning: Breaking Changes
+- API: Remove v1alpha4 API Version (#9939)
+- Dependency: Bump to controller-runtime v0.17.0 & k8s.io v0.29 (#9964)
+- Logging: Make event recorder names consistent for KCP and topology controller (#9862)
+- MachinePool: Set MachinePool feature flag default to true + Beta (#10141)
+- KCP/MachineSet: Objects generated by KCP, MachineSets and MachinePools will now consistently use machine name (#9833)
+
+## :sparkles: New Features
+- API: Enable kubeadm feature gates mutation (#10154)
+- clusterctl: Add k0smotron provider (#9991)
+- clusterctl: Add public function to create new CRD migrator (#10075)
+- Control-plane: Allow mutation of all fields that should be mutable (#9871)
+- MachineDeployment: Add MachineSetReady condition to MachineDeployment (#9262)
+- MachineDeployment: Taint nodes with PreferNoSchedule during rollouts (#10223)
+- MachineHealthCheck: implement annotation to manually mark machines for remediation (#10202)
```

Comment:
Not 100% sure if these qualify as highlight but these are nice features :-).

Reply:
Agree, also included in my highlight section proposal below